File: erp-mobile-app/src/utils/permissionModules.ts
/**
 * Map mobile screen/module id to backend permission module code (role_permissions.module).
 * Used for permission-based dashboard and navigation guard.
 */
import type { Screen } from '../types';

const SCREEN_TO_MODULE: Record<string, string> = {
  sales: 'sales',
  purchase: 'purchase', // singular to match DB
  pos: 'pos',          // match DB pos module
  rental: 'rentals',
  studio: 'studio',
  accounts: 'ledger',  // map accounts screen to ledger module
  expense: 'payments',  // fallback to payments as there is no expenses module
  products: 'inventory', // products screen maps to inventory module
  inventory: 'inventory',
  contacts: 'contacts',
  reports: 'reports',
  packing: 'sales',    // packing list is part of sales/wholesale workflow
  ledger: 'ledger',    // customer ledger read-only
  settings: 'settings',
  dashboard: 'reports', // dashboard uses reports permission
  home: 'reports',     // home always shown; "More" grid still filtered
};

export function getPermissionModuleForScreen(screen: Screen): string | null {
  return SCREEN_TO_MODULE[screen] ?? null;
}

export function screenRequiresPermission(screen: Screen): boolean {
  return screen !== 'login' && screen !== 'branch-selection' && screen !== 'home' && getPermissionModuleForScreen(screen) != null;
}

File: erp-flutter-v2/lib/core/permissions/permission_modules.dart
// Screen → role_permissions.module mapping (erp-mobile-app/src/utils/permissionModules.ts)

enum ErpScreen {
  login,
  branchSelection,
  home,
  dashboard,
  sales,
  purchase,
  rental,
  studio,
  accounts,
  expense,
  inventory,
  products,
  pos,
  contacts,
  reports,
  packing,
  ledger,
  settings,
}

const Map<ErpScreen, String> screenToPermissionModule = {
  ErpScreen.sales: 'sales',
  ErpScreen.purchase: 'purchase',
  ErpScreen.pos: 'pos',
  ErpScreen.rental: 'rentals',
  ErpScreen.studio: 'studio',
  ErpScreen.accounts: 'ledger',
  ErpScreen.expense: 'payments',
  ErpScreen.products: 'inventory',
  ErpScreen.inventory: 'inventory',
  ErpScreen.contacts: 'contacts',
  ErpScreen.reports: 'reports',
  ErpScreen.packing: 'sales',
  ErpScreen.ledger: 'ledger',
  ErpScreen.settings: 'settings',
  ErpScreen.dashboard: 'reports',
  ErpScreen.home: 'reports',
};

String? permissionModuleForScreen(ErpScreen screen) =>
    screenToPermissionModule[screen];

bool screenSkipsModuleViewPermission(ErpScreen screen) =>
    screen == ErpScreen.settings;
